projects/media-viewer/src/lib/toolbar/side-bar/outline-view/outline-view.component.ts

```diff
@@ -0,0 +1,139 @@
+import * as pdfjsLib from 'pdfjs-dist';
+import * as pdfjsOutlineViewer from 'pdfjs-dist/lib/web/pdf_outline_viewer';
+import {
+  AfterContentInit, Component, ElementRef, Input, OnChanges, SimpleChanges, ViewChild,
+  ViewEncapsulation
+} from '@angular/core';
+import { PdfJsWrapperFactory } from '../../../viewers/pdf-viewer/pdf-js/pdf-js-wrapper.provider';
+
+@Component({
+  selector: 'mv-outline-view',
+  templateUrl: './outline-view.component.html',
+  styleUrls: ['../../../styles/main.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class OutlineViewComponent implements OnChanges, AfterContentInit {
+
+  @Input() url: string;
+  @ViewChild('outlineContainer') outlineContainer: ElementRef;
+
+  private pdfDocumentOutline: pdfjsOutlineViewer.PDFOutlineViewer;
+  private outline: {};
+
+  constructor(private readonly pdfJsWrapperFactory: PdfJsWrapperFactory) {
+  }
+
+  async ngAfterContentInit() {
+    this.renderOutlineView();
+  }
+
+  async ngOnChanges(changes: SimpleChanges) {
+    if (changes.url) {
+      this.outline = {};
+      await this.renderOutlineView();
+    }
+  }
+
+  async renderOutlineView() {
+    this.pdfDocumentOutline = this.pdfJsWrapperFactory.createDocumentOutline(this.outlineContainer);
+    const pdf = await pdfjsLib.getDocument({
+      url: this.url,
+      cMapUrl: 'assets/minified/cmaps',
+      cMapPacked: true,
+      withCredentials: true
+    });
+
+    this.outline = await pdf.getOutline();
+    this.render({ outline: this.outline });
+  }
+
+  bindOutlineLink(element, { url, newWindow, dest }) {
```

Comment:
Proper type info here would be nice, wrap that second argument into an interface perhaps